builtin/providers/profitbricks/resource_profitbricks_datacenter.go

```diff
@@ -0,0 +1,202 @@
+package profitbricks
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/profitbricks/profitbricks-sdk-go"
+	"log"
+	"runtime"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func resourceProfitBricksDatacenter() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceProfitBricksDatacenterCreate,
+		Read:   resourceProfitBricksDatacenterRead,
+		Update: resourceProfitBricksDatacenterUpdate,
+		Delete: resourceProfitBricksDatacenterDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+
+			//Datacenter parameters
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"location": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceProfitBricksDatacenterCreate(d *schema.ResourceData, meta interface{}) error {
+	username, password, _ := getCredentials(meta)
```

Comment:
this feels onerous for someone extending - I think there should be a helper func that takes care of this

```
conn := getProfitBricksConnection(meta)
```